### PR TITLE
feat: multi-threded tests for the MPL

### PR DESCRIPTION
**Pull request by ajewellamz**
*Issue #, if available:*

*Description of changes:*

*Squash/merge commit message, if applicable:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: StandardLibrary/src/CallMany.dfy [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  Depending on the platform, the number of parallel threads is either `threads` or `outerIters`
+*/
+
+module {:extern "CallMany"} CallMany {
+  import opened StandardLibrary
+  import opened Wrappers
+  import opened StandardLibrary.UInt
+
+  trait {:termination false} Callee {
+    method call(inner : uint32, outer : uint32)
```
Comment (seebees) [inline StandardLibrary/src/CallMany.dfy:27]: `inner` and `outer` could use better names.
Something about the threadiness.

File: StandardLibrary/src/CallMany.dfy [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+include "./UInt.dfy"
+
+/*
+  Call a single function many times in many threads
+
+  CallMany(callee : Callee, innerIters : uint32, outerIters : uint32, threads : uint32)
+  This is effectively 
+
+  for i n 0..outerIters
+     for j in 0..innerIters
```
Comment (seebees) [inline StandardLibrary/src/CallMany.dfy:14]: Comment above about the thread bounadry.

**Review by seebees — CHANGES_REQUESTED** [review]

File: StandardLibrary/src/CallMany.dfy [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+       callee(j, i)
+
+  but multi threaded.
+  Depending on the platform, the number of parallel threads is either `threads` or `outerIters`
```
Comment (robin-aws) [inline StandardLibrary/src/CallMany.dfy:18]: Not following without a bit more detail

File: StandardLibrary/test/CallMany.dfy [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+      if count < UINT32_MAX {
+        count := count + 1; // not technically thread safe, but usually works
+      }
+    }
+  }
+
+  method {:test} TestBasic() {
+    var c := new MyCallee();
+    CallMany.CallMany(c, 2, 3, 2);
+    expect c.count == 6;
```
Comment (robin-aws) [inline StandardLibrary/test/CallMany.dfy:33]: There's no way expecting one exact value isn't going to be flaky as heck given the above. :) The best you can do is `expect min <= c.count <= max`.

And even if everything was thread safe, if the behaviour of the `CallMany` above is 2 threads doing 6 total increments, shouldn't you expect 12?

**Review by robin-aws — CHANGES_REQUESTED** [review]

**Review by seebees — APPROVED** [review]
LGTM

**Pull request merged**